src/Rule/Date.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator\Rule;
+
+use Closure;
+use Attribute;
+use InvalidArgumentException;
+use Yiisoft\Validator\WhenInterface;
+use Yiisoft\Validator\DumpedRuleInterface;
+use Yiisoft\Validator\SkipOnErrorInterface;
+use Yiisoft\Validator\SkipOnEmptyInterface;
+use Yiisoft\Validator\Rule\Trait\WhenTrait;
+use Yiisoft\Validator\RuleHandlerInterface;
+use Yiisoft\Validator\Rule\Trait\SkipOnEmptyTrait;
+use Yiisoft\Validator\Rule\Trait\SkipOnErrorTrait;
+
+/**
+ * @see DateHandler
+ * @psalm-import-type WhenType from WhenInterface
+ */
+#[Attribute(Attribute::TARGET_PROPERTY | Attribute::IS_REPEATABLE)]
+final class Date implements DumpedRuleInterface, SkipOnErrorInterface, WhenInterface, SkipOnEmptyInterface
+{
+    use SkipOnEmptyTrait;
+    use SkipOnErrorTrait;
+    use WhenTrait;
+
+    /**
+     * @var string The regular expression used to validate the value. See
+     * {@link https://www.regular-expressions.info/email.html}.
+     * @psalm-var non-empty-string
+     */
+    private string $pattern;
+
+    /**
+     * @psalm-var non-empty-string
+     */
+    private string $format;
+    /**
+     * @param string $format The format of the date.
```

Comment:
Need to describe what could be used in the format string or provide a link to PHP manual.